### PR TITLE
Adding insights data to metadataContext

### PR DESCRIPTION
**Pull request by ruks**
### Purpose
- Adding insights data to metadataContext

### Issues

Fixes #

### Automation tests
 - Unit tests added: Yes/No
 - Integration tests added: Yes/No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [ ] Assigned 'Type' label
- [ ] Assigned the project
- [ ] Validated respective github issues
- [ ] Assigned milestone to the github issue(s)

### CODE REVIEW

File: enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/api/RestAPI.java [hunk 1]
```diff
@@ -146,6 +147,9 @@ public String init(Api api) {
                             api.getEndpointSecurity().getSandBoxSecurityInfo()));
         }
 
+        org.wso2.choreo.connect.enforcer.commons.model.ChoreoComponentInfo choreoComponentInfo = getChoreoComponentInfo(
```
Comment (VirajSalaka) [inline enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/api/RestAPI.java:150]: Shall we import the class

File: enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/api/RestAPI.java [hunk 2]
```diff
@@ -156,12 +160,26 @@ public String init(Api api) {
...
-                .environmentName(api.getEnvironmentName()).build();
+                .environmentName(api.getEnvironmentName())
+                .choreoComponentInfo(choreoComponentInfo)
+                .build();
 
         initFilters();
         return basePath;
     }
 
+    private static org.wso2.choreo.connect.enforcer.commons.model.ChoreoComponentInfo getChoreoComponentInfo(
```
Comment (VirajSalaka) [inline enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/api/RestAPI.java:171]: same here

File: enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/api/RestAPI.java [hunk 3]
```diff
@@ -156,12 +160,26 @@ public String init(Api api) {
...
+            Api api) {
+        ChoreoComponentInfo infoProto = api.getChoreoComponentInfo();
+        org.wso2.choreo.connect.enforcer.commons.model.ChoreoComponentInfo choreoComponentInfo =
+                new org.wso2.choreo.connect.enforcer.commons.model.ChoreoComponentInfo();
```
Comment (VirajSalaka) [inline enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/api/RestAPI.java:175]: Shall we handle the possibility of choreoComponentInfo being null

File: enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/analytics/AnalyticsFilter.java [hunk 4]
```diff
@@ -212,6 +164,80 @@ public void handleFailureRequest(RequestContext requestContext) {
         }
     }
 
+    private void addInsightsMetaData(RequestContext requestContext) {
+        String apiName = requestContext.getMatchedAPI().getName();
+        String apiVersion = requestContext.getMatchedAPI().getVersion();
+        String apiType = requestContext.getMatchedAPI().getApiType();
+        String projectId = requestContext.getMatchedAPI().getChoreoComponentInfo().getProjectID();
```
Comment (VirajSalaka) [inline enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/analytics/AnalyticsFilter.java:171]: I guess we shoud check the possibility of this property being null and assign some default value.

File: enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/api/RestAPI.java (hunk 1 above)
Comment (ruks) [reply enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/api/RestAPI.java:150]: a class with the same name already imported. Hence had to improve with the fully qualified class name.

File: enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/api/RestAPI.java (hunk 2 above)
Comment (ruks) [reply enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/api/RestAPI.java:171]: a class with the same name already imported. Hence had to improve with the fully qualified class name.

**Review by VirajSalaka — APPROVED** [review]

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(EastUS) cluster : dev-deployment-v2 : 20241015.8

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : dev-deployment-v2 : 20241015.8

**Comment by choreo-cicd** [comment]
[succeeded]  : dev-deployment-v2 : 20241015.8

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : stage-deployment-v2 : 20241017.1

**Comment by choreo-cicd** [comment]
[failed] Controlplane cluster : stage-deployment-v2 : 20241017.1

**Comment by choreo-cicd** [comment]
[failed] Dataplane(EastUS) cluster : stage-deployment-v2 : 20241017.1

**Comment by choreo-cicd** [comment]
[] Dataplane(EastUS) cluster : stage-deployment-v2 : 20241017.2

**Comment by choreo-cicd** [comment]
[] Dataplane(NorthEU) cluster : stage-deployment-v2 : 20241017.2

**Comment by choreo-cicd** [comment]
[] Controlplane cluster : stage-deployment-v2 : 20241017.2

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : stage-deployment-v2 : 20241017.4

**Comment by choreo-cicd** [comment]
[failed] Controlplane cluster : stage-deployment-v2 : 20241017.4

**Comment by choreo-cicd** [comment]
[] Controlplane cluster : stage-deployment-v2 : 20241018.2

**Comment by choreo-cicd** [comment]
[] Dataplane(NorthEU) cluster : stage-deployment-v2 : 20241018.2

**Comment by choreo-cicd** [comment]
[] Dataplane(EastUS) cluster : stage-deployment-v2 : 20241018.2

**Comment by choreo-cicd** [comment]
[] Dataplane(EastUS) cluster : prod-deployment-v2 : 20241021.3

**Comment by choreo-cicd** [comment]
[] Controlplane cluster : prod-deployment-v2 : 20241021.3

**Comment by choreo-cicd** [comment]
[] Dataplane(NorthEU) cluster : prod-deployment-v2 : 20241021.3

**Comment by choreo-cicd** [comment]
[] Dataplane(CDP2-WU2) cluster : prod-deployment-v2 : 20241021.3

**Pull request merged**